python/clean_export_aggregate.py

```diff
@@ -0,0 +1,53 @@
+import pandas as pd
+import sys
+from sqlalchemy import create_engine
+
+BUILD_ENGINE = sys.argv[1]
+
+template_lookup = {
+    "aggregate_cdta_2020": "CDTA",
+    "aggregate_block_2020": "CensusBlocks",
+    "aggregate_tract_2020": "CensusTracts",
+    "aggregate_councildst_2010": "CityCouncil",
+    "aggregate_commntydst_2010": "CommunityDistricts",
+    "aggregate_nta_2020": "NTA",
+}
+
+
+def read_aggregate_template(name: str):
+    base = pd.read_csv(
+        f"data/agg_template/{template_lookup[name]}.csv", dtype=str)
+    base.set_index(get_index_columns(name),
+                   inplace=True, verify_integrity=True)
+    base.drop(columns=["OBJECTID", "boro"], axis=1, inplace=True)
+    return base
+
+
+def get_index_columns(name: str):
+    index_name = {
+        "aggregate_cdta_2020": "cdta2020",
+        "aggregate_block_2020": "bctcb2020",
+        "aggregate_tract_2020": "bct2020",
+        "aggregate_councildst_2010": "councildst",
+        "aggregate_commntydst_2010": "commntydst",
+        "aggregate_nta_2020": "nta2020",
+    }
+    return index_name[name]
+
+
+if __name__ == "__main__":
+
+    table = sys.argv[2]
+
+    engine = create_engine(BUILD_ENGINE)
+    base = read_aggregate_template(table)
+    df = pd.read_sql(f"""SELECT * FROM {table}""", con=engine)
+
+    df.dropna(axis=0, subset=[get_index_columns(table)], inplace=True)
+    df.set_index(get_index_columns(table), inplace=True, verify_integrity=True)
+
+    df_concat = pd.concat([base, df], axis=1)
```

Comment:
This is good and is the right approach. I have some thoughts about which pandas functions to try. We could use concat, merge or join as [this blog post](https://towardsdatascience.com/python-pandas-dataframe-join-merge-and-concatenate-84985c29ef78) outlines. 
My preference is to use merge for two reasons. The first is that it's what we use extensively in EDDE and there's a benefit in consistency for readability. The second is that you pass `left_index=True` and `right_index=True` which makes the code more readable as it's clear you're joining on the indicies. 
Concat allows us to join on columns as well as rows, which is not functionality we need here.
There's also an argument for join as it's the same language as SQL which makes the pipeline a bit more readable. 

I don't have strong feelings on which we choose, just wanted to raise the question

Reply:
wait I thought the more common use for this type of join is concat see [here](https://github.com/NYCPlanning/db-equitable-development-tool/search?q=concat). Also I like this because it ensures unique geography per row as a results where as merge/join does not guarantee that.

Reply:
Oh oops maybe we should have had more standardization there. I always used [merge in EDDE](https://github.com/NYCPlanning/db-equitable-development-tool/search?q=merge). 
I think that all three would ensure a unique geography by row right? As long as the geography data in `data/agg_template` has unique geographies and we use an inner join. And the `verify_integrity` flag should ensure unique geographies in `base`. 
I think any of the three are ok, it would be better to have more standardization for this operation but kind of too late now

Reply:
I definitely wasn't consistent in my own use lol as exhibited in the change in units case. I do think concat works in this case since it is designed to do this kind of join on index along. For multiple columns join and also not necessarily unique records, I do also prefer merge or join but I almost think of the kind operation happening here as "dumb join" if that makes any sense. @mbh329 not sure I understand your point though left join does not ensuring all the no match would be zeros they would be null. They would also create duplicates if they are multiple matches in the right dataframes.

Reply:
It is a left join, pandas has the merge, join and concat that can all do left joins and we are trying to figure out which is best

Reply:
I assumed that there would be no duplicate geographies in either. If the aggregate table comes from a groupby we can be confident there should be no duplicate geographies right? And in the big picture those nulls do get sent to zero. 
What do you mean by "dumb join"? I thought this was just the typical left join use in EDDE for many indicators

Reply:
I guess the set as index step does ensure uniqueness. It is dumb in the sense that it is almost not a "join" more a "concat" in the sense that two dataframes with same index therefore number of rows being smooshed together compared to something that requires more sophisticated matching. I do consider some of the EDDE join fell into the same category.